include/deal.II/grid/tria_accessor.h

```diff
@@ -1871,9 +1871,7 @@ class TriaAccessor : public TriaAccessorBase<structdim, dim, spacedim>
   clear_children() const;
 
 private:
-  template <int, int>
-  DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
-  friend class Triangulation;
+  friend class Triangulation<dim, spacedim>;
```

Comment:
Note that this really was a bug: `dim` and `spacedim` in the `requires` clause refer to the template arguments *of the surrounding class* because the template arguments of the `friend` declaration weren't named. Your solution is correct; another solution would have been to say this instead:
```
  template <int dim2,spacedim2>
  DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim2, spacedim2>))
  class Triangulation;
```
instead.

But I like your solution better.